set.mm

```diff
@@ -147288,24 +147690,66 @@ of an open range of nonnegative integers (of length equal to the length of
+    VBWDWIWAELWDWGAWHBWDAVSAVCFZVNVPXEVRVPAAGCTVFOQVDWDBVSBVCFZVNVRXFVPVRBBACTV
+    FQQVDVGRVHVI $.
+
+  $( Obsolete proof of ~ pfxpfx as of 12-Oct-2022.  (Contributed by Alexander
+     van der Vekens, 7-Apr-2018.)  (Proof modification is discouraged.)
+     (New usage is discouraged.) $)
+  swrd0swrd0OLD $p |- ( ( W e. Word V /\ N e. ( 0 ... ( # ` W ) )
        /\ L e. ( 0 ... N ) ) -> ( ( W substr <. 0 , N >. ) substr <. 0 , L >. )
                                 = ( W substr <. 0 , L >. ) ) $=
     ( cword wcel cc0 chash cfv cfz co w3a cop csubstr wceq 3simpa elfznn0 0elfz
-    wa cn0 syl 3ad2ant2 simp3 jca swrdswrd0 sylc ) DCEFZBGDHIZJKFZAGBJKZFZLZUGU
-    ISGUJFZUKSDGBMNKGAMZNKDUNNKOUGUIUKPULUMUKUIUGUMUKUIBTFUMBUHQBRUAUBUGUIUKUCU
-    DGABCDUEUF $.
+    wa cn0 syl 3ad2ant2 simp3 jca swrdswrd0OLD sylc ) DCEFZBGDHIZJKFZAGBJKZFZLZ
+    UGUISGUJFZUKSDGBMNKGAMZNKDUNNKOUGUIUKPULUMUKUIUGUMUKUIBTFUMBUHQBRUAUBUGUIUK
+    UCUDGABCDUEUF $.
 
-  $( A prefix of a prefix with the same length is the prefix.  (Contributed by
-     AV, 5-Apr-2018.)  (Proof shortened by AV, 14-Oct-2018.) $)
-  swrd0swrdid $p |- ( ( W e. Word V /\ N e. ( 0 ... ( # ` W ) ) )
+  $( A prefix of a prefix.  (Contributed by Alexander van der Vekens,
+     7-Apr-2018.)  (Revised by AV, 8-May-2020.) $)
+  pfxpfx $p |- ( ( W e. Word V /\ N e. ( 0 ... ( # ` W ) )
+                     /\ L e. ( 0 ... N ) )
+                   -> ( ( W prefix N ) prefix L ) = ( W prefix L ) ) $=
+    ( wcel cc0 cfz co w3a cpfx cop csubstr wa wceq elfznn0 anim2i pfxval nn0cnd
+    cn0 syl cword chash cfv caddc 3adant3 oveq1d cmin simp1 simp2 3ad2ant2 3jca
+    subid1d eqcomd adantl oveq2d eleq2d biimp3a pfxswrd addid2d opeq2d 3ad2ant3
+    0elfz sylc 3adant2 eqtr4d 3eqtrd ) DCUAZEZBFDUBUCZGHEZAFBGHZEZIZDBJHZAJHDFB
+    KLHZAJHZDFFAUDHZKZLHZDAJHZVMVNVOAJVMVHBSEZMZVNVONVHVJWBVLVJWAVHBVIOZPUEDBVG
+    QTUFVMVHVJFVKEZIAFBFUGHZGHZEZVPVSNVMVHVJWDVHVJVLUHVHVJVLUIVJVHWDVLVJWAWDWCB
+    VBTUJUKVHVJVLWGVHVJMZVKWFAWHBWEFGVJBWENVHVJWEBVJBVJBWCRULUMUNUOUPUQAFBCDURV
+    CVMVSDFAKZLHZVTVLVHVSWJNVJVLVRWIDLVLVQAFVLAVLAABOZRUSUTUOVAVMVHASEZMZVTWJNV
+    HVLWMVJVLWLVHWKPVDDAVGQTVEVF $.
+
+  $( Obsolete proof of ~ pfxpfxid as of 12-Oct-2022.  (Contributed by AV,
+     5-Apr-2018.)  (Proof shortened by AV, 14-Oct-2018.)
+     (Proof modification is discouraged.)  (New usage is discouraged.) $)
+  swrd0swrdidOLD $p |- ( ( W e. Word V /\ N e. ( 0 ... ( # ` W ) ) )
                         -> ( ( W substr <. 0 , N >. ) substr <. 0 , N >. )
                            = ( W substr <. 0 , N >. ) ) $=
     ( cword wcel cc0 chash cfv cfz co cop csubstr wceq cn0 elfznn0 nn0fz0 sylib
-    adantl swrd0swrd0 mpd3an3 ) CBDEZAFCGHZIJEZAFAIJEZCFAKZLJZUELJUFMUCUDUAUCAN
-    EUDAUBOAPQRAABCST $.
+    adantl swrd0swrd0OLD mpd3an3 ) CBDEZAFCGHZIJEZAFAIJEZCFAKZLJZUELJUFMUCUDUAU
+    CANEUDAUBOAPQRAABCST $.
+
+  $( A prefix of a prefix with the same length is the prefix.  (Contributed by
```

Comment:
is the ORIGINAL prefix. In other words, the operation "prefix of length N" is idempotent.

Reply:
Done